packages/twenty-server/src/engine/core-modules/user/utils/format-user-vars.util.ts

```diff
@@ -0,0 +1,27 @@
+import { KeyValuePair } from 'src/engine/core-modules/key-value-pair/key-value-pair.entity';
+
+export const formatUserVars = (
+  userVars: Pick<KeyValuePair, 'key' | 'value' | 'userId' | 'workspaceId'>[],
+): Map<string, JSON> => {
+  const workspaceUserVarMap = new Map<string, any>();
+  const userUserVarMap = new Map<string, any>();
+  const userWorkspaceUserVarMap = new Map<string, any>();
+
+  for (const { key, value, userId, workspaceId } of userVars) {
+    if (!userId && workspaceId) {
```

Comment:
not a big fan of elseif pattern:

if (xxx) {
...
continue
} 
is clearer to me. here we are missing the else case for example